File: src/java_package/Ellipse.java
package java_package;

public class Ellipse extends Curve {
    private double a; // ellipse's a coefficient
    private double b; // ellipse's b coefficient

    public Ellipse(double a, double b) {
        this.a = a;
        this.b = b;
    }

    @Override
    double calculateY(double x) {
        return Math.sqrt(1 - (Math.pow(x, 2) / Math.pow(a, 2))) * b;
    }
}

File: src/kotlin_package/Ellipse.kt
package kotlin_package

import kotlin.math.pow
import kotlin.math.sqrt

class Ellipse(private var a: Double, private var b: Double) : Curve() {
    @Override
    override fun calculateY(x: Double): Double {
        return sqrt(1 - (x.pow(2) / a.pow(2))) * b
    }
}